keyext.slicing/src/main/java/org/key_project/slicing/DependencyTracker.java

```diff
@@ -218,15 +256,19 @@ public void ruleApplied(ProofEvent e) {
         // record any rules added by this rule application
         for (NodeReplacement newNode : ruleAppInfo.getReplacementNodes()) {
             for (NoPosTacletApp newRule : newNode.getNode().getLocalIntroducedRules()) {
-                if (newRule.rule() instanceof Taclet
-                        && ((Taclet) newRule.rule()).getAddedBy() == n) {
+                if (newRule.rule() instanceof Taclet taclet && taclet.getAddedBy() == n) {
                     AddedRule ruleNode = new AddedRule(newRule.rule().name().toString());
                     output.add(ruleNode);
                     dynamicRules.put((Taclet) newRule.rule(), ruleNode);
                 }
             }
         }
 
+        // skip further tracking if disabled
+        if (!SlicingSettingsProvider.getSlicingSettings().getAlwaysTrack()) {
+            return;
+        }
+
```

Comment:
> If online tracking is enabled, a suggestion would be to register also to the AutoModeListener. When automode starts: save the current open leaf nodes, and when the automode stop event arrives update the tracking information from that point onwards.

This won't work, the tracker requires all previous nodes to be tracked.

Reply:
> If I see it right, the recording of the added rules can also be done first when required if tracking is switched on.

Yes. I think it can be done similar to #3311. I will wait until that PR is merged to avoid merge conflicts.